Tools/TestWebKitAPI/Tests/WebKitCocoa/WKURLSchemeHandler-1.mm

```diff
@@ -1731,3 +1731,24 @@ - (void)userContentController:(WKUserContentController *)userContentController d
     [webView loadRequest:[NSURLRequest requestWithURL:[NSURL URLWithString:@"test://webkit.org/main.html"]]];
     TestWebKitAPI::Util::run(&done);
 }
+
+TEST(URLSchemeHandler, RedirectFromHTTPToHandledScheme)
+{
+    TestWebKitAPI::HTTPServer server({
+        { "/redirect.html"_s, { 302, {{ "Location"_s, "testing://destination.html"_s }}, "redirecting..."_s } }
+    });
+
+    __block bool schemeHandledCalled = false;
+    RetainPtr schemeHandler = adoptNS([TestURLSchemeHandler new]);
+    [schemeHandler setStartURLSchemeTaskHandler:^(WKWebView *, id<WKURLSchemeTask> task) {
+        schemeHandledCalled = true;
+    }];
+
+    RetainPtr configuration = adoptNS([[WKWebViewConfiguration alloc] init]);
+    [configuration setURLSchemeHandler:schemeHandler.get() forURLScheme:@"testing"];
+
+    RetainPtr webView = adoptNS([[WKWebView alloc] initWithFrame:CGRectMake(0, 0, 800, 600) configuration:configuration.get()]);
+    [webView loadRequest:server.request("/redirect.html"_s)];
+
+    TestWebKitAPI::Util::run(&schemeHandledCalled);
+}
```

Comment:
It lost my comment here. I was asking for 1) a second test that goes the opposite direction, i.e. custom protocol to HTTP(S), and 2) a test case with another 3xx redirect such as 301.

Reply:
Redirection from custom scheme to HTTP(S) is something I'm trying to support here. It's certainly not what broke and I have no idea if this used to work or not.

Reply:
Better to know than not. I'm not asking for further behavior changes, just testing that we don't regress something else.

Reply:
My change can't change the behavior of redirections to HTTP(s) since it is gated on having a custom scheme handler for the destination URL scheme.

Reply:
I haven't been able to write such a test yet. I asked Brady if it's possible or not. As far as I can tell, when the custom scheme handler triggers a redirect, it has to provide the response for the redirected URL as well. So we can't really redirect from custom scheme to HTTP it seems.

Reply:
I just got confirmation from Brady that redirecting from custom scheme to HTTP is just not possible with the current API.

> You can re-write the URL with a redirect, but then need to keep handling the task yourself.

So the scheme handler can trigger a server redirect but then it has to handle that redirect itself, it cannot just say "redirect to HTTP and stop using me" for this navigation.